### DISCUSSION TITLE
"not defined" error when variable within a loop

### DISCUSSION THREAD

**Discussion by dpinol** [JuliaPluto/PlutoTest.jl#7] [Issues] [answered]
Hi,
is there any workaround for this issue? It works find if the tested variable is not declared within a loop

thanks
```
begin
	for i in 1:1
		a=3
		@test a==4
	end
end
```

> UndefVarError: a not defined
> 
>     top-level scope@:0
>     eval@boot.jl:360[inlined]
>     seval@notebook.jl:402[inlined]
>     var"#onestep_light#8"(::Module, ::typeof(PlutoTest.onestep_light), ::Expr)@notebook.jl:421
>     macro expansion@notebook.jl:480[inlined]
>     top-level scope@Local: 4

**Comment by disberd** [comment, marked as answer]
I found that interpolating the variable inside the test expression is needed to make local scope variables work.

![image](https://user-images.githubusercontent.com/12846528/130989286-a8cf0a6f-b62a-4e7b-a5d1-2e8ec560dc7b.png)

**Reply by dralletje** [reply]
Until we do some smart stuff, this is still necessary.
It's possible to make it "just" work with normal variables, but it's going to take a couple of hours and make the code a bunch more complex 😅

**Reply by dralletje** [reply]
Couple of hours taken, and now local variables do work!
It's in the latest github version, if you could test it that would be great, in any case I'll ping here when it's released too!

**Comment by fonsp** [comment]
Unfortunately this is beyond me! It would be great if someone with more metaprogramming experience could take a look at this!

**Reply by dralletje** [reply]
> Writes Pluto

> Asks for someone with more metaprogramming experience

🤡
